lib/jsdom/living/helpers/selectors.js

```diff
@@ -1,28 +1,40 @@
 "use strict";
 
+const DOMException = require("domexception");
+const nwsapi = require("nwsapi");
 const idlUtils = require("../generated/utils");
-const nwmatcher = require("nwmatcher/src/nwmatcher-noqsa");
 
 exports.matchesDontThrow = (elImpl, selector) => {
   const document = elImpl._ownerDocument;
 
-  if (!document._nwmatcherDontThrow) {
-    document._nwmatcherDontThrow = nwmatcher({ document });
-    document._nwmatcherDontThrow.configure({ UNIQUE_ID: false, LOGERRORS: false, VERBOSITY: false, SVG_LCASE: true });
+  if (!document._nwsapiDontThrow) {
+    document._nwsapiDontThrow = nwsapi({
+      document,
+      DOMException
+    });
+    document._nwsapiDontThrow.configure({
+      LOGERRORS: false,
+      VERBOSITY: false
+    });
   }
 
-  return document._nwmatcherDontThrow.match(idlUtils.wrapperForImpl(elImpl), selector);
+  return document._nwsapiDontThrow.match(selector, idlUtils.wrapperForImpl(elImpl));
 };
 
-// nwmatcher gets `document.documentElement` at creation-time, so we have to initialize lazily, since in the initial
+// nwsapi gets `document.documentElement` at creation-time, so we have to initialize lazily, since in the initial
 // stages of Document initialization, there is no documentElement present yet.
-exports.addNwmatcher = parentNode => {
+exports.addNwsapi = parentNode => {
   const document = parentNode._ownerDocument;
 
-  if (!document._nwmatcher) {
-    document._nwmatcher = nwmatcher({ document });
-    document._nwmatcher.configure({ UNIQUE_ID: false, LOGERRORS: false, SVG_LCASE: true });
+  if (!document._nwsapi) {
+    document._nwsapi = nwsapi({
```

Comment:
Do we have to put it on the document like this? Can we just store the reference in private state?

Reply:
This seems worth exploring separately, but with my time being rather limited right now I'd like to keep this pull request to just the changes needed for the switch.